src/main/java/org/openrewrite/java/migrate/lang/ReplaceRuntimeFinalizer.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2023 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.openrewrite.java.migrate.lang;
+
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.Value;
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Preconditions;
+import org.openrewrite.Recipe;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.java.JavaIsoVisitor;
+import org.openrewrite.java.JavaTemplate;
+import org.openrewrite.java.search.UsesType;
+import org.openrewrite.java.tree.J;
+import org.openrewrite.java.tree.JavaType;
+import org.openrewrite.java.tree.TypeUtils;
+
+import java.util.Collections;
+import java.util.Objects;
+import java.util.Set;
+
+@Value
+@EqualsAndHashCode(callSuper = true)
+@Getter
+public class ReplaceRuntimeFinalizer extends Recipe {
+
+    private static final String METHOD_PATTERN = "runFinalizersOnExit";
```

Comment:
It would be more idiomatic to add two `MethodMatcher`s for `java.lang.Runtime runFinalizersOnExit(boolean)` and `java.lang.System runFinalizersOnExit(boolean)`. That has the advantage that they will also match if these methods were imported statically.